Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIEventCocoa.mm

```diff
@@ -99,11 +99,12 @@
     m_frameIdentifier = frameIdentifier;
     m_listeners.append(listener);
 
-    RefPtr webFrame = WebProcess::singleton().webFrame(frameIdentifier);
-    RefPtr webExtensionControllerProxy = webFrame->page()->webExtensionControllerProxy();
-
     if (!hasExtensionContext()) {
-        if (webExtensionControllerProxy->inTestingMode()) {
+        RefPtr webFrame = WebProcess::singleton().webFrame(m_frameIdentifier);
+        RefPtr webPage = webFrame ? webFrame->page() : nullptr;
```

Comment:
could we just early return if we don't get a `webPage`?

Reply:
Early returns would be fine too, but no need to log as this could happen in normal situations.